pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiOffloadTest.java

```diff
@@ -290,6 +293,21 @@ public void testSetNamespaceOffloadPolicies() throws Exception {
         assertEquals(admin.namespaces().getOffloadPolicies(myNamespace), policies);
     }
 
+    @Test
+    public void testSetNamespaceOffloadPoliciesFailByReadOnly() throws Exception {
+        pulsar.getConfigurationMetadataStore().put(NamespaceResources.POLICIES_READONLY_FLAG_PATH, "0".getBytes(),
+                Optional.empty()).join();
+        try {
+            admin.namespaces().setOffloadThresholdInSeconds(myNamespace, 300);
+            fail("set offload threshold should fail when ns policies is readonly");
+        } catch (Exception ex){
+            // ignore.
+        }
+        // cleanup.
+        pulsar.getConfigurationMetadataStore().delete(NamespaceResources.POLICIES_READONLY_FLAG_PATH,
```

Comment:
this must be done in a `finally` block otherwise, If the test fails, the current state is not cleaned

Reply:
Already fixed. Thanks